federatedscope/core/monitors/metric_calculator.py

```diff
@@ -64,22 +64,27 @@ def _check_and_parse(self, ctx):
         if torch is not None and isinstance(y_prob, torch.Tensor):
             y_prob = y_prob.detach().cpu().numpy()
 
-        if y_true.ndim == 1:
-            y_true = np.expand_dims(y_true, axis=-1)
-        if y_prob.ndim == 2:
-            y_prob = np.expand_dims(y_prob, axis=-1)
-
-        y_pred = np.argmax(y_prob, axis=1)
-
-        # check shape and type
-        if not isinstance(y_true, np.ndarray):
-            raise RuntimeError('Type not support!')
-        if not y_true.shape == y_pred.shape:
-            raise RuntimeError('Shape not match!')
-        if not y_true.ndim == 2:
-            raise RuntimeError(
-                'y_true must be 2-dim arrray, {}-dim given'.format(
-                    y_true.ndim))
+        if 'regression' in ctx.cfg.model.task.lower():
```

Comment:
All functions `eval_xxx` like `eval_rmse` supports multi-task regression, and should we use `expand_dim` when the task is not multi-task regression?